src/coreclr/jit/importer.cpp

```diff
@@ -13859,6 +13863,13 @@ GenTree* Compiler::impInlineFetchArg(unsigned lclNum, InlArgInfo* inlArgInfo, In
 
             lvaTable[tmpNum].lvType = lclTyp;
 
+            if (!argCanBeModified)
+            {
+                assert(lvaTable[tmpNum].lvSingleDef == 0);
+                lvaTable[tmpNum].lvSingleDef = 1;
+                JITDUMP("Marked V%02u as a single def temp\n", tmpNum);
+            }
+
```

Comment:
Can you fuse this `if` with the one below?

Reply:
Won't the code be messier then?

Reply:
Looks better to me, something like:

```C++
// If arg can't be modified, mark it as single def.
// For ref types, determine the class of the arg temp.
if (!argCanBeModified)
{
    assert(lvaTable[tmpNum].lvSingleDef == 0);
    lvaTable[tmpNum].lvSingleDef = 1;
    JITDUMP("Marked V%02u as a single def temp\n", tmpNum);
    if (lclTyp == TYP_REF)
    {
         lvaSetClass(tmpNum, argNode, lclInfo.lclVerTypeInfo.GetClassHandleForObjRef());
    }
}
else 
{
    if (lclTyp == TYP_REF)
    {
        // Arg might be modified, use the declared type of
        // the argument.
        lvaSetClass(tmpNum, lclInfo.lclVerTypeInfo.GetClassHandleForObjRef());
    }
}
```

Reply:
Changed